lib/actuators/src/ContServo.cpp

```diff
@@ -0,0 +1,31 @@
+#include "ContServo.h"
+
+using namespace Actuator;
+
+ContServo::ContServo::ContServo(PinName pin, float max_speed, std::chrono::duration<float> max_pulse,
+                                std::chrono::duration<float> min_pulse, std::chrono::duration<float> period)
+    : m_pwm(pin), m_abs_max_speed(std::abs(max_speed)), m_speed(0) {
+  // Set initial condition of PWM
+  m_pwm.period(period.count());
+  m_pwm = 0.0;
+
+  // Set max and min pulse widths
+  m_max_pulse = max_pulse;
+  m_min_pulse = min_pulse;
+}
+
+ContServo::ContServo::ContServo(const Config &config)
+    : ContServo(config.pwmPin, config.max_speed, config.max_pulse, config.min_pulse, config.period) {}
+
+float ContServo::ContServo::getMaxSpeed(void) const {
+  return m_abs_max_speed;
+}
+
+void ContServo::ContServo::setValue(float speed) {
+  int sign = (speed >= 0) ? 1 : -1;
+  m_speed  = (std::abs(speed) < m_abs_max_speed) ? speed : m_abs_max_speed * sign;
```

Comment:
do you do anything with m_speed? It doesn't seem like this value is ever accessed in another of the other functions

Reply:
not too sure, i dindnt write this. maybe raise an issue abt it or something?

Reply:
hmm good point, I think originally if there was a getter function like getSpeed() then it would return that. Looks like it's not necessary if we don't need that

Reply:
okay i will fix